src/graphql/account.rs

```diff
@@ -418,6 +418,49 @@ impl AccountMutation {
         update_jwt_expires_in(expires_in)?;
         Ok(time)
     }
+
+    /// Retrieves the user's language selection by username.
+    #[graphql(guard = "RoleGuard::new(super::Role::SystemAdministrator)
+        .or(RoleGuard::new(super::Role::SecurityAdministrator))
+        .or(RoleGuard::new(super::Role::SecurityManager))
+        .or(RoleGuard::new(super::Role::SecurityMonitor))")]
+    async fn language(&self, ctx: &Context<'_>, username: String) -> Result<Option<String>> {
+        let store = crate::graphql::get_store(ctx).await?;
+        let map = store.account_map();
+
+        if let Some(account) = map.get(&username).map_err(|e| e.to_string())? {
+            Ok(account.language)
+        } else {
+            Err("Account not found".into())
+        }
+    }
+
+    /// Updates only the user's language setting, regardless of the user's role.
+    #[graphql(guard = "RoleGuard::new(super::Role::SystemAdministrator)
+        .or(RoleGuard::new(super::Role::SecurityAdministrator))
+        .or(RoleGuard::new(super::Role::SecurityManager))
+        .or(RoleGuard::new(super::Role::SecurityMonitor))")]
+    async fn update_language(
+        &self,
+        ctx: &Context<'_>,
+        username: String,
+        language: Option<UpdateLanguage>,
+    ) -> Result<String> {
+        let store = crate::graphql::get_store(ctx).await?;
+        let map = store.account_map();
+        let language = language.map(|d| (d.old, d.new));
```

Comment:
This allows users to change their language to `None`.  Is this what you intended?

Reply:
I intended to prevent users from assigning a value of `None`, so I added code to return an error if `None` is provided. However, I'm uncertain if this is the best approach. Could you please review the code and let me know if there are any areas that need improvement?

Reply:
My question was not prefect. It should be rephrased as:

This allows users to input `None` as the new language, and if they do, the language remains unchanged without any notification.

Reply:
I have modified the code to throw an error if `language.new` is `None`. Additionally, I have adjusted the argument type to `UpdateLanguage` in order to ensure that `None` cannot be passed as an argument.